src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -91,6 +98,11 @@ internal static ManagedIdentitySource GetManagedIdentitySource(ILoggerAdapter lo
                 logger?.Info("[Managed Identity] Azure Arc detected.");
                 return ManagedIdentitySource.AzureArc;
             }
+            else if (await ImdsV2ManagedIdentitySource.GetCsrMetadataAsync(_requestContext).ConfigureAwait(false))
```

Comment:
Can this source be only detected by sending a request? Should it be part of this method to detect the source? Maybe we can move this under selecting between IMDS V2 or IMDS v1? This change breaks the public API to get the source.

Reply:
yes and yes. Gladwin and Bogdan have signed off on this design.